<?xml version="1.0" ?>
<sdf version="1.4">
  <world name="vrc_task_1">
    <!-- place user camera -->
    <gui>
        <camera name="user_camera">
        <pose>-16 0 4 0 0 0</pose>
      </camera>
    </gui>
	<scene>
      <ambient>0.2 0.2 0.2 1.0</ambient>
      <grid>false</grid>
      <sky>
      <clouds>
          <speed>16</speed>
		  <humidity>100</humidity>
		  <mean_size>10</mean_size>
        </clouds>
      </sky>
      </scene>
	  
    <model name="ground_plane">
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>500 500</size>
            </plane>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>1</mu>
                <mu2>1</mu2>
              </ode>
            </friction>
          </surface>
        </collision>

		 <visual name="LFLY">
          <pose>141 -100 0 0 0 -1.5707</pose>
          <cast_shadows>false</cast_shadows>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>3661 2589</size>
            </plane>
          </geometry>
          <material>
          <script>
            <uri>file://materials/scripts</uri>
            <uri>file://materials/textures</uri>
            <name>Cessna/LFLY</name>
          </script>
        </material>
        </visual>
      </link>
	  </model>
	  
    <physics type="ode">
      <gravity>0 0 -9.81</gravity>
      <ode>
        <solver>
          <use_dynamic_moi_rescaling>true</use_dynamic_moi_rescaling>
          <type>quick</type>
          <iters>50</iters>
	  <!-- Must be 1.0 ! Not any above number -->
          <sor>1.0</sor>
        </solver>
        <constraints>
          <cfm>0.0</cfm>
          <erp>0.2</erp>
          <contact_max_correcting_vel>100.0</contact_max_correcting_vel>
          <contact_surface_layer>0.0</contact_surface_layer>
        </constraints>
      </ode>
      <real_time_update_rate>1000</real_time_update_rate>
      <max_step_size>0.001</max_step_size>
    </physics>

    <include>
      <uri>model://sun</uri>
    </include>
	
    <plugin name="ardupilot_sitl_gazebo_plugin" filename="libardupilot_sitl_gazebo_plugin.so">
	<UAV_MODEL>cessna</UAV_MODEL>
    <NB_SERVOS_MOTOR_SPEED>5</NB_SERVOS_MOTOR_SPEED>
	</plugin>
  </world>
</sdf>
